src/uu/split/src/split.rs

```diff
@@ -668,17 +668,38 @@ impl Strategy {
     }
 }
 
-/// Parse the suffix type, start and length from the command-line arguments.
-/// Determine if the output file names suffix is allowed to auto-widen,
-/// i.e. go beyond suffix_length, when more output files need to be written into.
-/// Suffix auto-widening rules are:
-/// - OFF when suffix length N is specified
-///   `-a N` or `--suffix-length=N`
-/// - OFF when suffix start number N is specified using long option with value
+/// Parse the suffix type, start and length from the command-line arguments
+/// as well suffix length auto-widening and auto-width scenarios
+///
+/// Suffix auto-widening: Determine if the output file names suffix is allowed to dynamically auto-widen,
+/// i.e. change (increase) suffix length dynamically as more files need to be written into.
+/// Suffix length auto-widening rules are (in the order they are applied):
+/// - ON by default
+/// - OFF when suffix start N is specified via long option with a value
 ///   `--numeric-suffixes=N` or `--hex-suffixes=N`
-/// - Exception to the above: ON with `-n`/`--number` option (N, K/N, l/N, l/K/N, r/N, r/K/N)
+/// - OFF when suffix length N is specified, except for N=0 (see edge cases below)
+///   `-a N` or `--suffix-length=N`
+/// - OFF if suffix length is auto pre-calculated (auto-width)
+///
+/// Suffix auto-width: Determine if the the output file names suffix length should be automatically pre-calculated
+/// based on number of files that need to written into, having number of files known upfront
+/// Suffix length auto pre-calculation rules:
+/// - Pre-calculate new suffix length when `-n`/`--number` option (N, K/N, l/N, l/K/N, r/N, r/K/N)
+///   is used, where N is number of chunks = number of files to write into
 ///   and suffix start < N number of files
-/// - ON when suffix start number is NOT specified
+///   as in `split --numeric-suffixes=1 --number=r/100 file`
+/// - Do NOT pre-calculate new suffix length otherwise, i.e. when
+///   suffix start >= N number of files
+///   as in `split --numeric-suffixes=100 --number=r/100 file`
+///   OR when suffix length N is specified, except for N=0 (see edge cases below)
+///   `-a N` or `--suffix-length=N`
+///
+/// Edge case:
+/// - If suffix length is specified as 0 AND `-n`/`--number` option used specifying number of files:
+///   set auto widening OFF AND auto pre-calculate required suffix length based on number of files needed
+/// - If suffix length is specified as 0 in any other situation
+///   keep auto widening ON and suffix length to default value
+///
```

Comment:
yep, I considered it. The best way is to separate suffix (or whole `Settings`) and `Strategy` into their own modules/files though, otherwise it would be cumbersome to pass this new `Suffix` type to `FilenameIterator` in filenames.rs. And if we keep the current way suffix values are passed to it, then there is no much simplification - just in calling `suffix_from`.
Would you be OK with that approach?